name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/TastytradePDFExtractor.java

```diff
@@ -0,0 +1,529 @@
+                else if (v.get("type").equals("to"))
+                    t.setType(AccountTransaction.Type.REMOVAL); 
+                t.setDateTime(asDate(v.get("date")));
+                t.setAmount(asAmount(v.get("amount"), "en", "US"));
+                t.setNote("Journal " + v.get("type") + " account " + v.get("account") + " " + v.get("sen"));
+            })
+            
+            .wrap(TransactionItem::new));
+    }
+    
+    private Money calculateFee(final ParsedData v)
+    {
+      BigDecimal amount = ExtractorUtils.convertToNumberBigDecimal(v.get("amount"), Values.Amount, "en", "US");
+      BigDecimal shares = ExtractorUtils.convertToNumberBigDecimal(v.get("shares"), Values.Amount, "en", "US");
+      BigDecimal price = ExtractorUtils.convertToNumberBigDecimal(v.get("price"), Values.Amount, "en", "US");
+      BigDecimal gross = price.multiply(shares);
+      BigDecimal fee = v.get("type").equals("SOLD") ? gross.subtract(amount) : amount.subtract(gross);
+      return Money.of("USD", asAmount(fee.toString(), "en", "US"));
+    }
+    
+    private <T extends Transaction<?>> void addInterestsSectionTransaction(T transaction, DocumentType type)
+    {
+        transaction
+            .section("date", "amount", "description", "details")
+            .match("^INTEREST[\\s]+(?<date>[\\d]{2}/[\\d]{2}/[\\d]{2})[\\s]+[A-Z][\\s]+(?<description>.*)[\\s]+(\\$)?(?<amount>[\\.,\\d]+).*$")
+            .match("^(?<details>.*$)")
+            .assign((t, v) -> {
+                ((name.abuchen.portfolio.model.Transaction) t).setDateTime(asDate(v.get("date")));
+                ((name.abuchen.portfolio.model.Transaction) t).setMonetaryAmount(Money.of("USD", asAmount(v.get("amount"), "en", "US")));
+                ((name.abuchen.portfolio.model.Transaction) t).setNote(v.get("description") + " " + v.get("details"));
+            });
+    }
+    
+    private void setTypeAndExpirationNote(BuySellEntry t, ParsedData v)
+    {
        BigDecimal quantity = ExtractorUtils.convertToNumberBigDecimal(v.get("quantity"), Values.Amount, "en", "US");
+
+        if (quantity.signum() == 1)
+        {
+            t.setType(PortfolioTransaction.Type.BUY);
+            t.setNote("Expired: Buy To Close");
+        }
+        else if (quantity.signum() == -1)
+        {
+            t.setType(PortfolioTransaction.Type.SELL);
+            t.setNote("Expired: Sell To Close");
+        }
+        else
+        {
+            throw new IllegalArgumentException(quantity.toString());
+        }
+    }
+
+    /**
+     * Convert date using US locale.
+     * 
+     * @param dateString
+     *            date with pattern M/d/yy
+     * @return converted date
+     */
+    private LocalDateTime asDate(String dateString)
+    {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("M/d/yy").withLocale(Locale.US);
+        return LocalDate.parse(dateString, formatter).atStartOfDay();
+    }
+}
```

Comment:
You can remove this part if you add the date format in [ExtractorUtils.java](https://github.com/portfolio-performance/portfolio/blob/master/name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/ExtractorUtils.java).
see --> DATE_FORMATTER_US ...

Reply:
Ah! Did not see this before.
Done ✔